### PR TITLE
Feature/sc 29053/implement getting a learning objects children

### PR DESCRIPTION
**Pull request by curlyLasagna**
Gets a learning-object's children. [Shortcut](https://app.shortcut.com/clarkcan/story/29053/implement-getting-a-learning-objects-children-in-clark-service)

Is it normal for the following animations to not be applied in service refactor?
`src/app/cube/details/components/splash/splash.component.ts`
```ts
@Component({
  selector: 'clark-details-splash',
  templateUrl: './splash.component.html',
  styleUrls: ['./splash.component.scss'],
  animations: [
    trigger('collection', [
      transition(':enter', [
        style({ opacity: 0, height: 0 }),
        animate('200ms ease', style({ opacity: 1, height: '*' }))
      ])
    ])
  ]
})
```

https://github.com/Cyber4All/clark-client/assets/44715649/dd2539c3-471e-41c8-9bda-fab71f90cd07

**Animation issue fixed**
See https://github.com/Cyber4All/clark-service/pull/138#discussion_r1638787031 for the solution

https://github.com/Cyber4All/clark-client/assets/44715649/de744fed-af1b-4507-9172-77a05c48b693

### CODE REVIEW

File: src/environments/route-copy.ts
Comment (johngonz0) [inline src/environments/route-copy.ts:1]: You can actually leave this file unchanged. It is for reference purposes for old routes. I would just revert the changes you made.

**Review by johngonz0 — CHANGES_REQUESTED** [review]
looks great, just a comment about the route-copy.ts file

**Review by JosueMolinaMorales — APPROVED** [review]

**Comment by mfranz1** [comment]
> Is it normal for the following animations to not be applied in service refactor?

I'm not sure I understand. The animations are not firing? Or what's the issue?

File: src/app/admin/components/change-author/change-author.component.ts [hunk 1]
```diff
@@ -37,11 +37,7 @@ export class ChangeAuthorComponent implements OnInit {
 
 
   async ngOnInit() {
-    const childrenUri = `${environment.apiURL}/users/${encodeURIComponent(
-      this.highlightedLearningObject.author.username
-      )}/learning-objects/${encodeURIComponent(
-      this.highlightedLearningObject.id
-    )}/children`;
+    const childrenUri = `${environment.apiURL}/learning-objects/${encodeURIComponent(this.highlightedLearningObject.id)}/children`;
```
Comment (mfranz1) [inline src/app/admin/components/change-author/change-author.component.ts:40]: Import route from `src/app/core/learning-object-module/...`

File: src/app/core/learning-object-module/learning-object/learning-object.service.ts [hunk 2]
```diff
@@ -160,15 +160,11 @@ export class LearningObjectService {
...
    * @param learningObjectId The object id of the learning object
    * @returns True if the learning object has children, false otherwise
    */
-  async doesLearningObjectHaveChildren(username: string, learningObjectId: string): Promise<boolean> {
-    const childrenUri = `${environment.apiURL}/users/${encodeURIComponent(username)
-      }/learning-objects/${encodeURIComponent(
-        learningObjectId
-      )}/children`;
+  async doesLearningObjectHaveChildren(learningObjectId: string): Promise<boolean> {
+    const childrenUri = `${environment.apiURL}/learning-objects/${encodeURIComponent(learningObjectId)}/children`;
```
Comment (mfranz1) [inline src/app/core/learning-object-module/learning-object/learning-object.service.ts:167]: Import route from `src/app/core/learning-object-module/...`

File: src/environments/route-copy.ts
Comment (mfranz1) [reply src/environments/route-copy.ts:1]: +1, not critical but when we really get into testing this helps just identify any old routes we had

**Review by mfranz1 — CHANGES_REQUESTED** [review]

**Comment by curlyLasagna** [comment]
> > Is it normal for the following animations to not be applied in service refactor?
> 
> I'm not sure I understand. The animations are not firing? Or what's the issue?

I should've specified that the video showed what I meant. When I click to make the sidebar appear with more details on the children in localhost, it doesn't apply the same animation as the one in production. Just wondering if it's normal

**Review by johngonz0 — APPROVED** [review]

**Review by shanioffe — APPROVED** [review]
Looks good!

File: src/app/core/learning-object-module/learning-object/learning-object.routes.ts [hunk 3]
```diff
@@ -7,12 +7,15 @@ export const LEARNING_OBJECT_ROUTES = {
      * @param learningObjectId the id of the learning object
      * @returns void
      */
+    GET_CHILDREN(learningObjectID: string) {
+        return `${environment.apiURL}/learning-objects/${encodeURIComponent(learningObjectID)}/children`;
+    },
```
Comment (mfranz1) [inline src/app/core/learning-object-module/learning-object/learning-object.routes.ts:12]: There's a block comment that belongs to the UPDATE_LEARNING_OBJECT_STATUS route function below. Can you please rearange and also create a block comment for this route. Just a generic description like the other block comment

**Review by mfranz1 — CHANGES_REQUESTED** [review]
Can you please fix the block comments in the routes

File: src/app/core/learning-object-module/learning-object/learning-object.routes.ts [hunk 4]
```diff
@@ -2,6 +2,15 @@ import { environment } from '../../../../environments/environment';
 import * as querystring from 'querystring';
 
 export const LEARNING_OBJECT_ROUTES = {
+    /**
+     * Request to get the children of a learning object
+     * @param learningObjectID the id of the learning object
```
Comment (mfranz1) [inline src/app/core/learning-object-module/learning-object/learning-object.routes.ts:7]:
```suggestion
     * @param learningObjectID the id of the parent learning object
```

**Review by mfranz1 — APPROVED** [review]

**Pull request merged**